tools/bazar/actions/BazarCartoAction.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+use YesWiki\Core\YesWikiAction;
+
+class BazarCartoAction extends YesWikiAction
+{
+    static $availableColors = array(
+        'red', 'darkred', 'lightred', 'orange', 'beige', 'green', 'darkgreen', 'lightgreen', 'blue', 'darkblue',
+        'lightblue', 'purple', 'darkpurple', 'pink', 'cadetblue', 'white', 'gray', 'lightgray', 'black',
+    );
+    
+    function formatArguments($arg)
+    {
+        // PROVIDERS
+        $provider = $_GET['provider']  ?? $arg['provider'] ?? $this->params->get('baz_provider');
+        $providerId = $arg['providerid'] ?? null ;
+        $providerPass = $arg['providerpass'] ?? null;
+        if (!empty($providerId) && !empty($providerPass)) {
+            if ($provider === 'MapBox') {
+                $providerCredentials = ', {id: \''.$providerId .'\', accessToken: \''.$providerPass.'\'}';
+            } else {
+                $providerCredentials = ', {app_id: \''.$providerId.'\',app_code: \''.$providerPass.'\'}';
+            }
+        } else {
+            $providerCredentials = '' ;
+        }
+
+        // ICONS
+        $iconField = $_GET['iconfield'] ?? $arg['iconfield'] ?? null;
+        $icon = $_GET['icon'] ?? $arg['icon'] ?? null;
+        if (!empty($icon)) {
+            $tabparam = $this->getMultipleParameters($icon, ',', '=');
+            if ($tabparam['fail'] != 1) {
+                if (count($tabparam) > 1 && !empty($iconField)) {
+                    // on inverse cle et valeur, pour pouvoir les reprendre facilement dans la carto
+                    foreach ($tabparam as $key=>$data) {
+                        $tabparam[$data] = $key;
+                    }
+                    $icon = $tabparam;
+                } else {
+                    $icon = trim($tabparam[0]);
+                }
+            } else {
+                exit('<div class="alert alert-danger">action bazarliste : le paramètre icon est mal rempli.<br />Il doit être de la forme icon="nomIcone1=valeur1, nomIcone2=valeur2"</div>');
+            }
+        } else {
+            $icon = $this->params->get('baz_marker_icon');
+        }
+
+        // COLORS
+        $colorField = $_GET['colorfield'] ?? $arg['colorfield'] ?? null;
+        $color = $_GET['color'] ?? $arg['color'];
```

Comment:
```suggestion
        $color = $_GET['color'] ?? $arg['color'] ?? null;
```
pour éviter des warnings